components/brave_wallet_ui/components/desktop/popup-modals/edit-visible-assets-modal/index.tsx

```diff
@@ -94,31 +117,56 @@ const EditVisibleAssetsModal = (props: Props) => {
           item.contractAddress.toLocaleLowerCase() === search.toLowerCase()
         )
       })
-      setFilteredWatchlist(filteredList)
+      setFilteredTokenList(filteredList)
     }
     setSearchValue(search)
-  }
+  }, [tokenList])
 
   const onClickAddCustomToken = () => {
-    if (foundToken !== '') {
-      onUpdateVisibleTokens(foundToken, true)
+    if (foundToken) {
+      onAddUserAsset(foundToken)
     } else {
-      onAddCustomToken(tokenName, tokenSymbol, tokenContractAddress, Number(tokenDecimals))
+      const newToken: TokenInfo = {
+        contractAddress: tokenContractAddress,
+        decimals: Number(tokenDecimals),
+        isErc20: true,
+        isErc721: false,
+        name: tokenName,
+        symbol: tokenSymbol,
+        logo: '',
+        visible: true
+      }
+      onAddUserAsset(newToken)
     }
+    setIsLoading(true)
   }
 
-  const onCheckWatchlistItem = (key: string, selected: boolean) => {
-    if (selected) {
-      const newList = [...selectedWatchlist, key]
-      setSelectedWatchlist(newList)
+  const isAssetSelected = (token: TokenInfo): boolean => {
+    const isUserToken = userVisibleTokensInfo.includes(token)
+    return isUserToken ? token.visible ?? false : false
+  }
```

Comment:
Is this a safe operation in JS? Would prefer using this instead:

```js
const isUserToken = userVisibleTokensInfo.map(e => e.contractAddress).includes(token.contractAddress)
```

Reply:
Done!